libs/ledger-live-common/src/exchange/swap/utils/inferRatesError.ts

```diff
@@ -0,0 +1,57 @@
+import BigNumber from "bignumber.js";
+import { getSwapAPIError } from "..";
+import {
+  SwapExchangeRateAmountTooHigh,
+  SwapExchangeRateAmountTooLow,
+  SwapGenericAPIError,
+} from "../../../errors";
+import { formatCurrencyUnit } from "@ledgerhq/coin-framework/currencies/formatCurrencyUnit";
+import { ExchangeRateV5Errors } from "../types";
+import { Unit } from "@ledgerhq/types-cryptoassets";
+
+export function inferError(response: ExchangeRateV5Errors, unitFrom: Unit): Error {
+  const isAMinMaxError =
+    "minAmountFrom" in response &&
+    "maxAmountFrom" in response &&
+    "amountRequested" in response &&
+    !!response.minAmountFrom &&
+    !!response.maxAmountFrom &&
+    !!response.amountRequested;
+
+  const isAnErrorCodeMessageError = "errorCode" in response && "errorMessage" in response;
+
+  if (isAMinMaxError) {
+    const isTooSmall = BigNumber(response.amountRequested).lt(response.minAmountFrom);
+
+    const minOrMaxError = isTooSmall
+      ? {
+          error: SwapExchangeRateAmountTooLow,
+          amount: response.minAmountFrom,
+          key: "minAmountFromFormatted",
+        }
+      : {
+          error: SwapExchangeRateAmountTooHigh,
+          amount: response.maxAmountFrom!,
+          key: "maxAmountFromFormatted",
+        };
+
+    return new minOrMaxError.error(undefined, {
+      [minOrMaxError.key]: formatCurrencyUnit(
+        unitFrom,
+        new BigNumber(minOrMaxError?.amount).times(BigNumber(10).pow(unitFrom.magnitude)),
+        {
+          alwaysShowSign: false,
+          disableRounding: true,
+          showCode: true,
+        },
+      ),
+      amount: new BigNumber(minOrMaxError.amount),
+    });
+  }
+
+  if (isAnErrorCodeMessageError) {
+    return getSwapAPIError(response.errorCode, response.errorMessage);
+  }
+
+  return new SwapGenericAPIError();
```

Comment:
There is a case where we are having a limit error but we do not know the limit value, neither if it is min or max.
Are we taking this into account? 
example: use a super small quantity with a swap that it is supported by DEX providers, for instance: eth vs tether, this limit is always smaller or bigger than the CEX ones, although there is not a proper amount.

Reply:
So I investigated this a bit, and it did exist, but there was no error message defined for that particular error. So the user would never get feedback as to why their quotes had errored. Not sure if we want to add one as part of this ticket?

Reply:
The behaviour is correct.
https://github.com/LedgerHQ/ledger-live/pull/2189